clients/application/src/main/java/org/eclipse/hono/client/application/ApplicationClientFactory.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.client.application;
+
+import java.util.function.Consumer;
+
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+
+/**
+ * A factory for creating clients for Hono's northbound APIs.
+ */
+public interface ApplicationClientFactory {
+
+    /**
+     * Creates a client for consuming data from Hono's northbound <em>Telemetry API</em>.
+     * <p>
+     * The messages passed in to the consumer will be acknowledged automatically if the consumer does not throw an
+     * exception.
+     *
+     * @param tenantId The tenant to consume data for.
+     * @param telemetryConsumer The handler to invoke with every message received.
+     * @param closeHandler The handler invoked when the consumer is closed (not by invoking
+     *            {@link MessageConsumer#close()}).
+     * @return A future that will complete with the consumer once the is ready. The future will fail if the the consumer
+     *         cannot be started.
+     * @throws NullPointerException if any of the parameters is {@code null}.
+     */
+    Future<MessageConsumer<TelemetryMessage>> createTelemetryConsumer(
+            String tenantId,
+            Consumer<TelemetryMessage> telemetryConsumer,
+            Handler<Throwable> closeHandler);
```

Comment:
I assume that the `Throwable` passed in can be `null` as well, right? If so, this should be documented FMPOV